File: Champions/Blitzcrank/Q.lua
Vector2 = require 'Vector2' -- include 2d vector lib 

function onStartCasting()
    spellAnimation("SPELL1", owner)
end

function onFinishCasting()
    local current = Vector2:new(owner.X, owner.Y)
    local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
    local range = to * 925
    local trueCoords = current + range
    addProjectile("RocketGrabMissile", trueCoords.x, trueCoords.y)
end

function applyEffects()
    local AP = owner:GetStats().AbilityPower.Total
    local damage = 25 + (spellLevel*55) + AP
    dealMagicalDamage(damage)

    if not isDead( getTarget() ) then
        addParticleTarget(owner, "Blitzcrank_Grapplin_tar.troy", getTarget(), 1, "L_HAND")
    	-- dash the user slightly in front of blitz
	    local current = Vector2:new(owner.X, owner.Y)
	    local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
    	local range = to * 50
	    local trueCoords = current + range

    	dashToLocation( getTarget(), trueCoords.x, trueCoords.y, projectileSpeed, true)
    end

    destroyProjectile()
end

function onUpdate(diff)

end

File: Champions/Blitzcrank/Q.cs
using System.Numerics;
using GameServerCore.Enums;
using static LeagueSandbox.GameServer.API.ApiFunctionManager;
using LeagueSandbox.GameServer.GameObjects.AttackableUnits;
using LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI;
using LeagueSandbox.GameServer.GameObjects.Missiles;
using LeagueSandbox.GameServer.GameObjects.Spells;
using LeagueSandbox.GameServer.Scripting.CSharp;
using GameServerCore;

namespace Spells
{
    public class RocketGrab : IGameScript
    {
        public void OnActivate(Champion owner)
        {
        }

        public void OnDeactivate(Champion owner)
        {
        }

        public void OnStartCasting(Champion owner, Spell spell, AttackableUnit target)
        {
            spell.SpellAnimation("SPELL1", owner);
        }

        public void OnFinishCasting(Champion owner, Spell spell, AttackableUnit target)
        {
            var current = new Vector2(owner.X, owner.Y);
            var to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
            var range = to * 925;
            var trueCoords = current + range;
            spell.AddProjectile("RocketGrabMissile", trueCoords.X, trueCoords.Y);
        }

        public void ApplyEffects(Champion owner, AttackableUnit target, Spell spell, Projectile projectile)
        {
            var ap = owner.Stats.AbilityPower.Total;
            var damage = new Damage(25 + spell.Level * 55 + ap, DamageType.DAMAGE_TYPE_MAGICAL, 
                DamageSource.DAMAGE_SOURCE_SPELL, false);
            target.TakeDamage(owner, damage);
            if (!target.IsDead)
            {
                AddParticleTarget(owner, "Blitzcrank_Grapplin_tar.troy", target, 1, "L_HAND");
                var current = new Vector2(owner.X, owner.Y);
                var to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
                var range = to * 50;
                var trueCoords = current + range;
                DashToLocation((ObjAiBase) target, trueCoords.X, trueCoords.Y,
                    spell.SpellData.MissileSpeed, true);
            }

            projectile.SetToRemove();
        }

        public void OnUpdate(double diff)
        {
        }
    }
}

